src/system/SystemLayerImplFreeRTOS.cpp

```diff
@@ -70,7 +70,9 @@ CHIP_ERROR LayerImplFreeRTOS::StartTimer(Clock::Timeout delay, TimerCompleteCall
         // HandleExpiredTimers() to re-start the timer.
         if (!mHandlingTimerComplete)
         {
-            ReturnErrorOnFailure(StartPlatformTimer(delay));
+            // NOTE(gmarcosb): You would think we could just propagate the failure here,
```

Comment:
StartTimer should be able to propagate failures, although perhaps as part of https://github.com/project-chip/connectedhomeip/issues/42025 it should simply die if it's unable to schedule